multicluster/apis/multicluster/v1alpha1/clusterproperty_types.go

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
@jianjuns It is not copied it, upstream only has the guide file, I wrote it

Reply:
Could you check what is this: https://github.com/kubernetes-sigs/about-api/blob/master/clusterproperty/api/v1alpha1/clusterproperty_types.go?

Have you checked what other MCS implementations do? If the CRD is not really finalized/stabilized, we need not to change now? @luolanzone

Reply:
@jianjuns I did not find out this, but the upstream CRD seems not completed. Do we just want to use this PR's? Or use this v1alpha1 or wait?

Reply:
@jianjuns  regarding the name change from ClusterClaim to ClusterProperty, it has no change since 2021 Nov. I feel the CRD and well known properties (cluster.clusterset.k8s.io and clusterset.k8s.io) are stable now. But the version is still v1apha1, I suppose we can on-hold this if you think it's better to wait a beta version.

Reply:
I checked the Submariner, it doesn't involve `ClusterProperty` but a `clusters.submariner.io` CRD including ClusterID and cluster info, will take a look at other CNIs and cloud providers.

Reply:
I just feel if no one uses `ClusterProperty` today and the CRD definition may even change in future, it may not help much for us to change now.

But on the other hand, it would be good if we can simplify our design and CRDs (e.g. removing ClusterSet).

BTW, you saw Quan proposed using antctl to convert Antrea CRs when promoting CRD versions to v1beta1. Could you check with Quan to see if we can use the same mechanism for the MC CRD change (rather than doing CRD mirroring)?

Reply:
Sure, I will check with Quan for the method, will also check how to simplify the design.

Reply:
@jianjuns I checked with @tnqn regarding the CLI tool to convert CRs, he mentioned the purpose is to handle the same CRD with different versions and only dummy update is needed for stored version change, which not suitable for our case. In our scenarios, we are actually handling different CRDs, different name and different groups. And he also suggested no need to add CRD conversion if there is no production case to cover. I feel we may check with Akshay first to see if we need to add a CRD conversion webhook, or just discard old CRD. 
The change from ClusterClaim to ClusterProperty, Quan also suggested to follow upstream design if there is no special requirement. I have sent a message on the sig-multicluster slack channel to understand the maturity of KEP-2145.

Reply:
I think we should not have any production deployment, but the question is what we do with existing ClusterSets created with old CRDs (even they are not in production). Do you mean we should ask users to recreate with the new CRDs?

And could you make clear if ClusterSet CRD is still needed in the latest KEP or not? I do not think we should just look at one CRD. Need to understand the complete design to decide what to do.

Reply:
ClusterSet is not defined in latest KEP, I am not sure if it's defined before. But I believe we still need a way to handle the basic information for a ClusterSet. I have created a page to track two CRDs including ClusterSet and MemberClusterAnnounce, we can discussion how to simplify on the MC sync meeting.

Reply:
For the old CRD, I believe we can use CRD mirroring (or CRD conversion?) to handle existing old CRs if deprecating it directly is not proper way. @bangqipropel is working on a separate PR for this.

Reply:
I checked upstream slack channel, they mentioned KEP-2149 will be in beta soon: `Promotion to beta is imminent, and at this time we’re not expecting any changes `